conda/cli/main_search.py

```diff
@@ -78,26 +80,22 @@ def configure_parser(sub_parsers):
         default=None,
     )
     p.add_argument(
-        "--spec",
-        action="store_true",
-        help="""Treat the regex argument as a package specification instead
-        (package_name[=version[=build]]).""",
+        'spec',
+        default='*',
+        nargs='?',
     )
+    # p.add_argument(
+    #     "--spec",
+    #     action="store_true",
+    #     help=SUPPRESS,
+    # )
```

Comment:
We should uncomment this block, but keep the help suppressed, since it's a no-op now.